core/runtime/binaryen/runtime_api/core_impl.cpp

```diff
@@ -25,7 +25,12 @@ namespace kagome::runtime::binaryen {
     BOOST_ASSERT(header_repo_ != nullptr);
   }
 
-  outcome::result<Version> CoreImpl::version() {
+  outcome::result<Version> CoreImpl::version(
+      std::optional<primitives::BlockHash> const &block_hash) {
+    if (!!block_hash) {
```

Comment:
just `if (block_hash) {` or `if (block_hash.has_value())` looks nicer IMO

Reply:
It's a habit remained from cx03, where shared_ptr in VS was in tr1 namespace and had only operator! for boolen output.